internal/manifests/mutate.go

```diff
@@ -95,3 +96,56 @@ func mutateServiceMonitor(existing, desired *monitoringv1.ServiceMonitor) {
 	// ServiceMonitor selector is immutable so we set this value only if
 	// a new object is going to be created
 }
+
+func mutatePodSpecForTLSEnablement(podSpec *corev1.PodSpec, serviceName string) {
```

Comment:
This function does not belong here. It is augmenting a pod spec if a feature flag is enabled. I tend to err this belongs more to `service_monitors.go`. In addition it is literarly configuring the pod spec to use the generated PKI. WDYT of calling such functions `configureServiceMonitorPKI()`.

This would look on the caller site like:
```go
if opt.EnableTLSServiceMonitors {
    configureServiceMonitorPKI(podSpec, serviceName)
}